Description: Recognize armel (linux-gnueabi) as a Linux arch, too.
Bug-Debian: https://bugs.debian.org/481928
Forwarded: no
Author: Miriam Ruiz <little_miry@yahoo.es>
Last-Update: 2015-12-18

--- a/configure.in
+++ b/configure.in
@@ -40,7 +40,7 @@
 
 dnl Aqui es donde se chequeara el SO, pero solo hay una opcion xD
 case "$target_os" in
-    linux-gnu)
+    linux-gnu*)
         case "$target_cpu" in
             powerpc*)
 	            COMMON_CFLAGS="-Wall -DTARGET_LinuxPPC -DTARGET_Linux"
