diff -ruN Fenix/common/files.c fenix-0.92a.dfsg1/common/files.c
--- Fenix/common/files.c	2007-04-12 22:07:56.000000000 +0000
+++ fenix-0.92a.dfsg1/common/files.c	2007-07-29 17:55:07.531111567 +0000
@@ -460,6 +460,31 @@
 	memset (f, 0, sizeof(file)) ;
 	strncpy (f->name, filename, MAX_PATH);
 
+	if (strcmp(filename, "-") == 0 && strchr(mode,'r'))
+	{
+		char buffer[1024];
+		size_t buflen;
+		f->type = F_FILE;
+		f->fp = tmpfile();
+		opened_files++;
+
+		while ((buflen = fread(buffer, 1, sizeof(buffer), stdin)))
+		{
+			fwrite(buffer, 1, buflen, f->fp);
+		}
+		fseek (f->fp, 0, SEEK_SET);
+
+		return f ;
+	}
+
+	if (strcmp(filename, "-") == 0 && strchr(mode,'w'))
+	{
+		f->type = F_STDFILE;
+		f->fp = stdout;
+		opened_files++;
+		return f ;
+	}
+
 	c = filename ;
 	for (n = c+strlen(c) ; n >= c ; n--)
 	{
