diff -ruN fenix0.92-0.92a.orig/configure fenix0.92-0.92a/configure
--- fenix0.92-0.92a.orig/configure	2007-04-13 18:19:02.000000000 +0000
+++ fenix0.92-0.92a/configure	2007-06-05 18:40:44.000000000 +0000
@@ -1629,7 +1629,7 @@
 
             *)
 	            COMMON_CFLAGS="-Wall -DTARGET_Linux"
-	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc -Wl,--dynamic-linker /lib/ld-linux.so.2"
+	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc"
                 ;;
         esac
 	    ;;
@@ -3176,7 +3176,7 @@
 if test "x$enable_debug" = "xyes"; then
 	CFLAGS="-g $INCLUDES"
 else
-	CFLAGS="-O3 -finline-functions -ffast-math $INCLUDES"
+	CFLAGS="-O2 -finline-functions -ffast-math $INCLUDES"
 fi
 
 
diff -ruN fenix0.92-0.92a.orig/configure.in fenix0.92-0.92a/configure.in
--- fenix0.92-0.92a.orig/configure.in	2007-04-13 18:19:02.000000000 +0000
+++ fenix0.92-0.92a/configure.in	2007-06-05 18:40:39.000000000 +0000
@@ -48,7 +48,7 @@
 
             *)
 	            COMMON_CFLAGS="-Wall -DTARGET_Linux"
-	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc -Wl,--dynamic-linker /lib/ld-linux.so.2"
+	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc"
                 ;;
         esac
 	    ;;
@@ -87,7 +87,7 @@
 if test "x$enable_debug" = "xyes"; then
 	CFLAGS="-g $INCLUDES"
 else
-	CFLAGS="-O3 -finline-functions -ffast-math $INCLUDES"
+	CFLAGS="-O2 -finline-functions -ffast-math $INCLUDES"
 fi
 
 dnl librerias para todos
