Description: Remove some weird options from the configure script.
Forwarded: no
Author: Miriam Ruiz <little_miry@yahoo.es>
Last-Update: 2015-12-18

--- a/configure.in
+++ b/configure.in
@@ -48,7 +48,7 @@
 
             *)
 	            COMMON_CFLAGS="-Wall -DTARGET_Linux"
-	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc -Wl,--dynamic-linker /lib/ld-linux.so.2"
+	            COMMON_LIBS="$COMMON_LIBS -Wl,-Bdynamic -lc"
                 ;;
         esac
 	    ;;
@@ -87,7 +87,7 @@
 if test "x$enable_debug" = "xyes"; then
 	CFLAGS="-g $INCLUDES"
 else
-	CFLAGS="-O3 -finline-functions -ffast-math $INCLUDES"
+	CFLAGS="-O2 -finline-functions -ffast-math $INCLUDES"
 fi
 
 dnl librerias para todos
