diff -ruN fenix-0.92a.raw/fxi/src/dcbr.c fenix-0.92a/fxi/src/dcbr.c
--- fenix-0.92a.raw/fxi/src/dcbr.c	2007-07-28 11:07:02.566318000 +0000
+++ fenix-0.92a/fxi/src/dcbr.c	2007-07-29 17:40:47.485362067 +0000
@@ -144,7 +144,7 @@
 	file * fp ;
 
 	/* check for existence of the DCB FILE */
-	if (!file_exists(filename)) return 0 ;
+	if (strcmp(filename,"-")!=0 && !file_exists(filename)) return 0 ;
 
 	fp = file_open (filename, "rb0") ;
 	if (!fp)
diff -ruN fenix-0.92a.raw/fxi/src/main.c fenix-0.92a/fxi/src/main.c
--- fenix-0.92a.raw/fxi/src/main.c	2007-07-28 11:06:34.620571000 +0000
+++ fenix-0.92a/fxi/src/main.c	2007-07-28 11:10:59.733140328 +0000
@@ -248,7 +248,7 @@
 
 		for (i = 1 ; i < argc ; i++)
 		{
-			if (argv[i][0] == '-')
+			if (argv[i][0] == '-' && argv[i][1])
 			{
 				j = 1 ;
 				while (argv[i][j])
