Description: Link against libgif
Author: Thibaut Gridel <tgridel@free.fr>
Bug-Debian: https://bugs.debian.org/732787
Forwarded: no
Last-Update: 2015-12-18

--- a/configure.in
+++ b/configure.in
@@ -115,7 +115,7 @@
 
 dnl para map
 if test "$map" = yes; then
-	AC_CHECK_LIB(ungif,DGifOpenFileName,LIBS="$LIBS",AC_MSG_ERROR(lungif library not found))
+	AC_CHECK_LIB(gif,DGifOpenFileName,LIBS="$LIBS",AC_MSG_ERROR(libgif library not found))
 fi
 
 dnl make these visible to all Makefiles
--- a/fpg/Makefile.am
+++ b/fpg/Makefile.am
@@ -5,7 +5,7 @@
 
 INCLUDES = $(COMMON_CFLAGS)
 
-fpg_LDADD=$(COMMON_LIBS) -lz  -lungif -lpng
+fpg_LDADD=$(COMMON_LIBS) -lz  -lgif -lpng
 
 SOURCE_FILES=fpg.c
 
--- a/map/Makefile.am
+++ b/map/Makefile.am
@@ -5,7 +5,7 @@
 
 INCLUDES = $(COMMON_CFLAGS)
 
-map_LDADD=$(COMMON_LIBS) -lz  -lungif -lpng
+map_LDADD=$(COMMON_LIBS) -lz  -lgif -lpng
 
 SOURCE_FILES=map.c
 
